src/istgt_lu_disk.c

```diff
@@ -5749,18 +5729,24 @@ replicate(ISTGT_LU_DISK *spec, ISTGT_LU_CMD_Ptr cmd, uint64_t offset, uint64_t n
 	while (1) {
 		MTX_LOCK(rcomm_cmd->mutex);
 		// check for status of rcomm_cmd
-		if (check_for_command_completion(spec, rcomm_cmd, cmd)) {
-			if (rcomm_cmd->status == -1)
-				rc = -1;
-			else
+		rc = check_for_command_completion(spec, rcomm_cmd, cmd);
+		if (rc) {
+			if (rc == 1)
 				rc = cmd->data_len = rcomm_cmd->data_len;
+			rcomm_cmd->state = CMD_EXECUTION_DONE;
 			put_to_mempool(&spec->rcommon_deadlist, rcomm_cmd);
+			MTX_UNLOCK(rcomm_cmd->mutex);
+
+			/*
+			 * NOTE: This is for debugging purpose only
+			 */
+			if (errno == ETIMEDOUT)
```

Comment:
'errno' is supposed to be of pthread_cond_timedwait. But, by the time execution reaches here, 'errno' might have got overwritten by other syscalls.
So, we need to store 'errno' into local variable.

Reply:
ok